Detector/DetCRD/src/Calorimeter/CRDEcal.cpp

```diff
@@ -0,0 +1,136 @@
+//====================================================================
+//  Detector description for CEPC Reference Detector ECal Barrel.
+//--------------------------------------------------------------------
+//
+//  Author: Fangyi Guo    email: guofangyi@ihep.ac.cn
+//  Ecal cosists of long crystal bar 1cm*1cm*~40cm 
+//  8 parts cover 2pi phi range. Inner radius, height, z-length can change from xml.
+//  In each part, crystal bar crosses in odd-even layer. 
+//
+//  v0r0: nothing but BGO crystal
+//
+//====================================================================
+#include "DD4hep/DetFactoryHelper.h"
+#include "XML/Layering.h"
+#include "XML/Utilities.h"
+#include "DDRec/DetectorData.h"
+#include "DDSegmentation/Segmentation.h"
+
+#define MYDEBUG(x) std::cout << __FILE__ << ":" << __LINE__ << ": " << x << std::endl;
+#define MYDEBUGVAL(x) std::cout << __FILE__ << ":" << __LINE__ << ": " << #x << ": " << x << std::endl;
+
+using dd4hep::rec::LayeredCalorimeterData;
+using namespace dd4hep;
+using namespace std;
+static dd4hep::Ref_t create_detector(dd4hep::Detector& theDetector,
+                                     xml_h e,
+                                     dd4hep::SensitiveDetector sens) {
+
+    xml_det_t x_det = e;
+
+    std::string det_name = x_det.nameStr();
```

Comment:
In CRDEcal.cpp, please make sure Tabs and Spaces are not mixed.
Please consider using 4 space for indentation.